jsk_pcl_ros/scripts/detect_graspable_poses_pcabase.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import rospy
+import sensor_msgs.point_cloud2 as pc2
+import tf.transformations
+
+from geometry_msgs.msg import Pose, PoseArray
+from jsk_topic_tools import ConnectionBasedTransport
+from sensor_msgs.msg import PointCloud2
+from sklearn.decomposition import PCA
+
+class DetectGraspablePosesPcabase(ConnectionBasedTransport):
+
+    def __init__(self):
+
+        rospy.init_node('detect_graspable_poses_pcabase')
+
+        rospy.logwarn("This node is experiential one.")
+
+        super(DetectGraspablePosesPcabase, self).__init__()
+
+        self.direction = rospy.get_param('~direction', 'x')
+        self.hand_size = rospy.get_param('~hand_size', 0.13) # maximum hand width
+        self.interval_m = rospy.get_param('~interval_m', '0.04') # interval of possible grasp
+
+        self.pub_targetPoses = self.advertise("~output/can_grasp_poses", PoseArray, queue_size=1)
```

Comment:
for `python` variables, it is common to use snake case `pub_target_poses`.
all variables should be written in snake case. (class is camel case)